### DISCUSSION TITLE
agg: keystroke overlay

### DISCUSSION THREAD

**Discussion by LangLangBart** [asciinema/discussions#172] [Ideas]
### Feature Request
Ability to add an overlay of keystrokes to the gif.

### Related tickets
- [asciinema/asciinema #105](https://github.com/asciinema/asciinema/issues/105#issuecomment-346974651)
> *v2 format has a place for keypresses*

### Screenshots/ Gifs
- Image from [tckmn/mkcast](https://github.com/tckmn/mkcast)

<img src="https://raw.githubusercontent.com/LangLangbart/ImagePool/770cbe461bb5130ddab492b41fb4f6211dd94c86/storage/2022-12-23_08-13-12_tckmn_mkcast.png" width="600">

- Gif with [Gifox](https://gifox.app/)
<img src="https://github.com/asciinema/discussions/assets/92653266/0d77e4db-ec85-43c6-ae22-291c5a103970" width="600">

**Comment by eduble** [comment]
Hi,

FYI, to solve this same problem I took a different approach and wrote a tool called [demoterm](https://github.com/drakkar-lig/demoterm) able to show keypresses on a reserved zone at the bottom of the terminal.
You can combine it with `asciinema` for recording, i.e., `asciinema rec -c demoterm`.

I used it for instance for the terminal recordings of this blog post:
https://walt-project.liglab.fr/blog/2025-11-auto-completion

It works fine on my Ubuntu Linux system (i.e., in `gnome-terminal`). There might be issues if the terminal does not support some of the escape sequences needed (e.g., for saving and restoring cursor position and color modes) though.
